File: node/common/src/client/models/WorkspaceDeletionRequest.ts
/* tslint:disable */
/**
 * Request to delete a workspace
 * @export
 * @interface WorkspaceDeletionRequest
 */
export interface WorkspaceDeletionRequest {
    /**
     * The App Id of this Theia.cloud instance. Request without a matching Id will be denied.
     * @type {string}
     * @memberof WorkspaceDeletionRequest
     */
    appId: string;
    /**
     * The user identification, usually the email address.
     * @type {string}
     * @memberof WorkspaceDeletionRequest
     */
    user: string;
    /**
     * The name of the workspace to delete.
     * @type {string}
     * @memberof WorkspaceDeletionRequest
     */
    workspaceName: string;
}

/**
 * Check if a given object implements the WorkspaceDeletionRequest interface.
 */
export function instanceOfWorkspaceDeletionRequest(value: object): boolean {
  let isInstance = true;
  isInstance = isInstance && 'appId' in value;
  isInstance = isInstance && 'user' in value;
  isInstance = isInstance && 'workspaceName' in value;

  return isInstance;
}

export function WorkspaceDeletionRequestFromJSON(json: any): WorkspaceDeletionRequest {
  return WorkspaceDeletionRequestFromJSONTyped(json, false);
}

export function WorkspaceDeletionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): WorkspaceDeletionRequest {
  if ((json === undefined) || (json === null)) {
    return json;
  }
  return {

    'appId': json['appId'],
    'user': json['user'],
    'workspaceName': json['workspaceName']
  };
}

export function WorkspaceDeletionRequestToJSON(value?: WorkspaceDeletionRequest | null): any {
  if (value === undefined) {
    return undefined;
  }
  if (value === null) {
    return null;
  }
  return {

    'appId': value.appId,
    'user': value.user,
    'workspaceName': value.workspaceName
  };
}

File: java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/workspace/WorkspaceDeletionRequest.java
/********************************************************************************
 * Copyright (C) 2022 EclipseSource and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * This Source Code may also be made available under the following Secondary
 * Licenses when the conditions for such availability set forth in the Eclipse
 * Public License v. 2.0 are satisfied: GNU General Public License, version 2
 * with the GNU Classpath Exception which is available at
 * https://www.gnu.org/software/classpath/license.html.
 *
 * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 ********************************************************************************/
package org.eclipse.theia.cloud.service.workspace;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.eclipse.theia.cloud.service.ServiceRequest;
import org.eclipse.theia.cloud.service.validation.Validate;
import org.eclipse.theia.cloud.service.validation.ValidationProblem;
import org.eclipse.theia.cloud.service.validation.ValidationResult;

@Schema(name = "WorkspaceDeletionRequest", description = "Request to delete a workspace")
public final class WorkspaceDeletionRequest extends ServiceRequest {
    public static final String KIND = "workspaceDeletionRequest";

    @Schema(description = "The user identification, usually the email address.", required = true)
    public String user;

    @Schema(description = "The name of the workspace to delete.", required = true)
    public String workspaceName;

    public WorkspaceDeletionRequest() {
	super(KIND);
    }

    public WorkspaceDeletionRequest(String appId, String user, String workspaceName) {
	super(KIND, appId);
	this.workspaceName = workspaceName;
	this.user = user;
    }

    @Override
    public String toString() {
	return "WorkspaceDeletionRequest [user=" + user + ", workspaceName=" + workspaceName + ", appId=" + appId
		+ ", kind=" + kind + "]";
    }

    @Override
    public ValidationResult validateDataFormat() {
	List<ValidationProblem> problems = new ArrayList<ValidationProblem>();
	validateServiceRequest(problems);
	Validate.user(user).ifPresent(problems::add);
	Validate.workspaceName(workspaceName).ifPresent(problems::add);
	return new ValidationResult(problems);
    }

}
